mule-extensions-api/src/main/java/org/mule/runtime/extension/api/manifest/ImmutableDescriberManifest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.manifest;
+
+import static java.util.Collections.unmodifiableMap;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import java.util.Map;
+
+/**
+ * Immutable implementation of {@link DescriberManifest}
+ *
+ * @since 1.0
+ */
+final class ImmutableDescriberManifest implements DescriberManifest
```

Comment:
ufff.. that one was a long debate including LG and other people... The main problem was that it should be possible to use the ext-api jar without the need of also having extension-support. That's the case for Studio, Mozart, etc.

There're use cases such as json/xml serialization (and some other I don't remember) in which at some point you need to create instances of such interfaces. It makes no sense to require the user to provide his own implementation. Having another jar with just a default implementation is overkill and didn't make sense either.

We finally decided to do this which nobody had a major concern with. We can discuss that offline if you want but at least for now this is consistent with the general approach.